js/clientJS/teamCalendar.js

```diff
@@ -0,0 +1,409 @@
+    if (dayOfWeek === "") return "";
+    return formatDate(getXDayOfWeek(date, daysOfWeek.indexOf(dayOfWeek)-1));
+}
+
+/**
+ * Main  function which generates and prints the calendar
+ * @param users the users of the team for which a row is generated
+ * @param currentDate the current date (is the start date of the calendar)
+ * @param sessionUser the user who is currently logged in
+ * @param teamId the id of the team
+ * @returns {Promise<void>}
+ */
+async function generateCalendar(users, currentDate, sessionUser, teamId) {
+    const calContainer = document.querySelector('.cal-container');
+    const today = new Date();
+
+    //Load data from monday to sunday
+    const teamData = await getDataFromTeam(getMondayOfWeek(currentDate), getSundayOfCurrentWeek(currentDate), teamId)
+
+    // Create the header of the calendar
+    $(".cal-container").html(daysOfWeek.map(day => {
+        const dateStr = getDateFromDay(currentDate, day);
+        const isToday = dateStr === formatDate(new Date());
+        const headerClass = isToday ? "grid-header-today" : "grid-header";
+        return (
+            `<div class="grid-item ${headerClass}">` +
+            `<div class="header-content-container">` +
+            `<p>${day}</p>` +
+            `<p class="date">${dateStr}</p>` +
+            `</div>` +
+            `</div>`
+        );
+    }).join(""));
+
+
+    const numRows = users.length;
+    const numCols = 8;
+
+    // Create grid layout with 8 columns and for each user a row
+    for (let i = 0; i < numRows; i++) {
+        for (let j = 0; j < numCols; j++) {
+            const gridItem = document.createElement('div');
+            const dateStr = getDateFromDay(currentDate, daysOfWeek[j]);
+            const isToday = dateStr === formatDate(new Date());
+            const headerClass = isToday ? "grid-item-today" : "grid-item";
+            gridItem.classList.add('grid-item');
+            gridItem.classList.add(headerClass);
+            const jDayObj = new Date(getXDayOfWeek(currentDate, j - 1))
+
+            // Add username to first column
+            if (j === 0) {
+                gridItem.innerText = users[i].username;
+                gridItem.classList.add('grid-item-username');
+            } else {
+                let innerHTML =
+                    '<div class="cell-content-container">' +
+                    '<div class="edit-content-row">' +
+                    `<input type="text" value="${users[i].username}" style="display: none" id="inputUsername"/>` +
+                    `<input type="text" value="${formatDate(getXDayOfWeek(currentDate, j - 1))}" style="display: none" id="inputDate"/>`;
+                //if(users[i].username === sessionUser) {
+                    if (jDayObj.getTime() >= today.getTime()) {
+                        innerHTML +=
+                            '<a class="edit">' +
+                            '<i class="ri-edit-fill"></i>' +
+                            '</a>';
+                    //}
```

Comment:
Is this simply for testing or checking if the calendar belongs to the logged in user? if the latter, you might want to add a comment that you intent to enable this functionality in the future. A @todo is probably a good idea so we don't miss it.

Reply:
Added TODO